mozregression/main.py

```diff
@@ -240,6 +251,10 @@ def cli(argv=None):
         test_runner = CommandTestRunner(fetch_config, options.command,
                                         persist=options.persist)
 
+    prefs = mozprofile.prefs.Preferences()
+    if options.preferences:
+        prefs.add_file(options.preferences)
+
```

Comment:
Well, prefs is our object that contain preferences. You added preferences from the --preferences file, but not from the --pref argument. Have a look at the implementation of the same thing for mozprofile: http://mozbase.readthedocs.org/en/latest/_modules/mozprofile/cli.html#MozProfileCLI.preferences

Also, could you create a function that will return the prefs object ? It will be easier to test. Last thing, could you make the --preferences a list (like for --pref) by adding nargs='*' ? I changed my mind on this after looking at the behaviour of mozprofile.